src/foam/nanos/auth/User.js

```diff
@@ -46,64 +47,10 @@ foam.CLASS({
       name: 'lastLogin',
       documentation: 'Date and time user last logged in.'
     },
-    {
-      class: 'String',
-      name: 'firstName',
-      tableWidth: 160,
-      documentation: 'First name of user.',
-      validateObj: function(firstName) {
-        if ( firstName.length > 70 ) {
-          return 'First name cannot exceed 70 characters.';
-        }
-
-        if ( /\d/.test(firstName) ) {
-          return 'First name cannot contain numbers.';
-        }
-      }
-    },
-    {
-      class: 'String',
-      name: 'middleName',
-      documentation: 'Middle name of user.',
-      validateObj: function(middleName) {
-        if ( middleName.length > 70 ) {
-          return 'Middle name cannot exceed 70 characters.';
-        }
-
-        if ( /\d/.test(middleName) ) {
-          return 'Middle name cannot contain numbers.';
-        }
-      }
-    },
-    {
-      class: 'String',
-      name: 'lastName',
-      documentation: 'Last name of user.',
-      tableWidth: 160,
-      validateObj: function(lastName) {
-        if ( lastName.length > 70 ) {
-          return 'Last name cannot exceed 70 characters.';
-        }
-
-        if ( /\d/.test(lastName) ) {
-          return 'Last name cannot contain numbers.';
-        }
-      }
-    },
-    {
-      class: 'String',
-      name: 'legalName',
-      documentation: 'Full legal name of user. Appends first, middle & last name.',
-      transient: true,
-      expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
-      },
-      javaGetter: `
-        return ! getMiddleName().equals("")
-          ? getFirstName() + " " + getMiddleName() + " " + getLastName()
-          : getFirstName() + " " + getLastName();
-      `,
-    },
+    'firstName',
```

Comment:
Why are we defining the properties again?

Reply:
Oh presumably for the java generation issue that Kirk was talking about.  Can we file an issue about that and then reference that issue from a //TODO comment here?

Reply:
Yes, it's because of the java generation problem. I'll do that now.

Reply:
Issue: https://github.com/foam-framework/foam2/issues/1529

Reply:
TODO comment added in 6c69df9